app/src/main/java/vn/sunasterisk/music_67/ui/search/SearchPresenter.kt

```diff
@@ -0,0 +1,44 @@
+package vn.sunasterisk.music_67.ui.search
+
+import vn.sunasterisk.music_67.data.model.Track
+import vn.sunasterisk.music_67.data.source.TracksDataSource
+import vn.sunasterisk.music_67.data.source.TracksRepository
+import vn.sunasterisk.music_67.service.PlayingSet
+import vn.sunasterisk.music_67.utils.StringUtils
+
+class SearchPresenter(private val repository: TracksRepository,
+                      private val view: SearchContract.View
+) : SearchContract.Presenter {
+    val tracks: List<Track> = ArrayList()
+    override fun searchTracks(text: String, action: String) {
+        when (action) {
+            ACTION_SEARCH_FROM_HOME -> handleSearchHome(text)
+            ACTION_SEARCH_FROM_NOW -> handleSearchNow(text)
+        }
+    }
+
+    private fun handleSearchHome(text: String) {
```

Comment:
format text: nếu là empty thì sao. Nếu chỉ toàn khoảng trắng thì sao

Reply:
e check newText bên trên r thì k cần check cái này nữa nhỉ

Reply:
okiee